crates/net/downloaders/src/bodies/concurrent.rs

```diff
@@ -48,430 +231,413 @@ where
     }
 }
 
-impl<Client, Consensus> BodyDownloader for ConcurrentDownloader<Client, Consensus>
+impl<B, C, DB> BodyDownloader for ConcurrentDownloader<B, C, DB>
 where
-    Client: BodiesClient,
-    Consensus: ConsensusTrait,
+    B: BodiesClient + 'static,
+    C: ConsensusTrait + 'static,
+    DB: Database,
 {
-    fn bodies_stream<'a, 'b, I>(&'a self, headers: I) -> DownloadStream<'a, BlockResponse>
-    where
-        I: IntoIterator<Item = &'b SealedHeader>,
-        <I as IntoIterator>::IntoIter: Send + 'b,
-        'b: 'a,
-    {
-        Box::pin(
-            stream::iter(headers.into_iter())
-                .chunks(self.batch_size)
-                .map(move |headers| {
-                    (move || self.fetch_bodies(headers.clone()))
-                        // We should not backoff from requests since the downloader does
-                        // not know what peer the request is being delivered to.
-                        // See: https://github.com/paradigmxyz/reth/issues/809
-                        .retry(
-                            ConstantBackoff::default()
-                                .with_delay(Duration::from_secs(0))
-                                .with_max_times(self.retries),
-                        )
-                })
-                .buffered(self.concurrency)
-                .map_ok(|blocks| stream::iter(blocks.into_iter()).map(Ok))
-                .try_flatten(),
-        )
+    /// Set a new download range (exclusive).
+    ///
+    /// This method will drain all queued bodies and re-request any
```

Comment:
this needs to be a bit more specific:
this ensures the queued bodies are in range, it does not drain all of them.

Reply:
will amend, but to reiterate my response above, the queue **is** drained